lib/end.go

```diff
@@ -89,3 +101,31 @@ func testEq(a, b []string) bool {
 	}
 	return true
 }
+
+func signACI(acipath, signaturepath, user string, flags []string) error {
+	if len(flags) == 0 {
+		flags = []string{"gpg", "--armor", "--yes"}
+	}
+	flags = append(flags, "--output", signaturepath, "--detach-sig", acipath)
+
+	var sucommand string
+	for _, flag := range flags {
+		if sucommand == "" {
+			sucommand = flag
+		} else {
+			sucommand = sucommand + " " + flag
+		}
+	}
+
+	var err error
+	if user == "" {
+		err = util.Exec("su", user, "-c", sucommand)
```

Comment:
What I had in mind was actually using `cmd.SysProcAttr.Credential` on the gpg command rather than calling out to su. I am not sure if there is any other initialization that I'm missing that would prevent gpg working that way as another user; can you try it?